dali/ft/filecopy.cpp

```diff
@@ -3876,27 +3876,20 @@ cost_type FileSprayer::updateSourceProperties()
                         // so query the first (and only) subfile
                         subfile = &superSrc->querySubFile(0);
                     }
-                    cost_type curReadCost = calcFileAccessCost(subfile, 0, curProgress.numReads);
-                    subfile->addAttrValue(getDFUQResultFieldName(DFUQRFnumDiskReads), curProgress.numReads);
-                    cost_type legacyReadCost = getLegacyReadCost(subfile->queryAttributes(), subfile);
-                    subfile->addAttrValue(getDFUQResultFieldName(DFUQRFreadCost), legacyReadCost + curReadCost);
-                    totalReadCost += curReadCost;
+                    totalReadCost += updateCostAndNumReads(subfile, curProgress.numReads);
                 }
                 else
                 {
                     // not sure if src superfile can have whichInput==-1 (but if so, this is best effort to calc cost)
                     totalReadCost += calcFileAccessCost(distributedSource, 0, curProgress.numReads);
                 }
             }
+            return updateCostAndNumReads(distributedSource, totalNumReads, totalReadCost);
```

Comment:
Probably reasonable to set these super props at this time, but they're going to be wrong if/when the super file is manipulated, e.g. a new subfile added or removed.

I think updateFileAttrs needs to be updated to cope with that.
They're also going to be wrong after o single subfile in an existing super is read ? i.e. what would update the super cost under those circumstances?

When the engines (e.g. thor) reads a super, it updates the sub file read costs in CMasterActivity::calcFileReadCostStats, I can't see that it updates the super costs though?

This PR may be okay as it is, but looks like there are problems with super cost info going stale in general?
@shamser - please comment, and if appropriate, open a new JIRA, reference here, and link it to this JIRA.

Reply:
A jira to revisit super file costs: https://hpccsystems.atlassian.net/browse/HPCC-32901